test/e2e-common.sh

```diff
@@ -330,6 +350,16 @@ function test_setup() {
 
   echo ">> Creating test resources (test/config/)"
   ko apply ${KO_FLAGS} -f test/config/ || return 1
+
+  # Depending on the flags `HTTP_ENABLED` and `AUTO_TLS_ENABLED`, apply the appropriate network configs
+  if (( HTTP_ENABLED )) &&  (( AUTO_TLS_ENABLED )); then
+    ko apply ${KO_FLAGS} -f test/config-network/config-network-http-https.yaml || return 1
+  elif (( HTTP_ENABLED )) && (( ! AUTO_TLS_ENABLED )); then
+    ko apply ${KO_FLAGS} -f test/config-network/config-network-http.yaml || return 1
```

Comment:
`http-only` might be more explicit than `http`.
Same with `https-only` vs. `https`.

Reply:
@JRBANCEL originally I was thinking of having flags that corresponded directly to the yaml values:
```
--http --> httpProtocol: "Enabled"
--no-http --> httpProtocol: "Disabled"
--auto-tls --> autoTLS: "Enabled"
--no-auto-tls --> autoTLS: "Disabled"
```
If we have flags `http-only` and `https-only`, do we need to introduce a third flag to indicate http and https?

Reply:
I would prefer not introducing auto TLS into this PR if we just want to test different HTTP and HTTPS setting because:
1. auto TLS needs more setup (DNS server, real domain, static IP, etc.)
2. If we run all E2E tests with auto TLS enabled, we can quickly run out of TLS certificates quota.

In this PR, I think we are targeting to test the scenario that HTTP server (https://github.com/knative/serving/blob/master/config/202-gateway.yaml#L27) is removed, and HTTPS server with certificates is added (https://github.com/knative/serving/blob/master/config/202-gateway.yaml#L34). So probably one flag is enough: --https-only. WDYT?

In addition, instead of customizing config-network.yaml, I think we just need to customize https://github.com/knative/serving/blob/master/config/202-gateway.yaml because `httpProtocol:Disabled` only works when we turn on either flag `autoTLS` or [`reconcileExternalGateway`](https://github.com/knative/serving/blob/bdde80f62580c30ae0179adaa145be093df5e2ef/config/config-istio.yaml#L65) which is bit more complex to set up.

Reply:
Thanks @ZhiminXiang ! I've added a gateway yaml for https only at `test/https-gateway.yaml`. I'm not sure where to put it so please let me know if you think there's a better destination.